enrolments/serializers.py

```diff
@@ -66,7 +65,7 @@ def get_families(self, obj):
 
 
 class EnrolmentSerializer(serializers.HyperlinkedModelSerializer):
-    session = SessionListSerializer()
+    session = SessionListSerializer(read_only=True)
     preferred_class = ClassListSerializer()
     enrolled_class = ClassListSerializer()
```

Comment:
it might not make a lot of sense to expect the frontend to provide the names of the classes & sessions when making this request - ideally, it would just pass the ID!

you might consider `updating session`, `enrolled_class`, and `preferred_class` to `PrimaryKeyRelatedFields`, and overriding `to_representation` (instead of `to_internal_value` & `update`) so that the GET endpoint still gets the serialized sessions & classes. check out the "original post" on [this stackoverflow answer](https://stackoverflow.com/a/52246232) (the updated one is a little much for us)

```suggestion
    session = serializers.PrimaryKeyRelatedField(read_only=True)
    preferred_class = serializers.PrimaryKeyRelatedField(queryset=Class.objects.all())
    enrolled_class = serializers.PrimaryKeyRelatedField(queryset=Class.objects.all())
```

Reply:
this solution is poggies, thank you for sharing